extensions/ql-vscode/src/databases/database-config.ts

```diff
@@ -0,0 +1,29 @@
+// Contains models for the data we want to store in the database config
+
+export interface DatabaseConfig {
+  remote: RemoteDatabaseConfig;
+}
+
+export interface RemoteDatabaseConfig {
+  repositoryLists: RemoteRepositoryList[];
+  owners: string[];
+  repositories: string[];
+}
+
+export interface RemoteRepositoryList {
+  name: string;
+  repositories: string[];
+}
+
+export function cloneDatabaseConfig(config: DatabaseConfig): DatabaseConfig {
+  return {
+    remote: {
+      repositoryLists: config.remote.repositoryLists.map((list) => ({
+        name: list.name,
+        repositories: list.repositories.slice(),
+      })),
+      owners: [...config.remote.owners],
+      repositories: [...config.remote.repositories],
+    }
+  };
```

Comment:
Possibly, this is more robust. The advantage is that if you change the shape of `DatabaseConfig` to add or remove properties, this function will not need to change the disadvantage is that any non-standard objects will lose their prototype.

```suggestion
  return JSON.parse(JSON.stringify(config));
```

Reply:
I generally avoid doing that because it's considered slow. 

I think the shape will change a little bit while we implement the feature but it won't change often because it's something that will be visible to the users.